be/extmap.c

```diff
@@ -874,20 +891,38 @@ emap_it_pack(struct m0_be_emap_cursor *it,
 	const struct m0_be_emap_seg *ext = &it->ec_seg;
 	struct m0_be_emap_key       *key = &it->ec_key;
 	struct m0_be_emap_rec       *rec = &it->ec_rec;
+	struct m0_buf rec_buf;
+	void *dst;
 
 	key->ek_prefix = ext->ee_pre;
 	key->ek_offset = ext->ee_ext.e_end;
 	m0_format_footer_update(key);
+
+	m0_buf_alloc(rec_buf, sizeof(struct m0_be_emap_rec) + ext->ee_di_cksum.b_nob);
+	dst = rec_buf.b_addr;
+
 	rec->er_start  = ext->ee_ext.e_start;
 	rec->er_value  = ext->ee_val;
 	m0_format_footer_update(rec);
+
+	/* I am not very sure of following code */
+	memcpy(dst, (void *)rec, offsetof(struct m0_be_emap_rec, er_value));
+	dst = dst + offsetof(struct m0_be_emap_rec, er_value);
+	if (ext->ee_di_cksum.b_nob != 0) {
+		memcpy(dst, ext->ee_di_cksum.b_addr, ext->ee_di_cksum.b_nob);
+		dst += ext->ee_di_cksum.b_nob;
+	}
+	memcpy(dst, rec->er_footer, sizeof(struct m0_format_footer));
```

Comment:
Please describe what you want to do.
What is the expected data schema in dst?

Reply:
@huanghua78 - please ignore this PR. I will close it and delete the branch. This was a sample code written to get the review if approach is correct. Though approach is correct but this code does not compile or tested. I will come up with another PR with working code.